dGame/dUtilities/Loot.cpp

```diff
@@ -137,10 +137,6 @@ std::unordered_map<LOT, int32_t> LootGenerator::RollLootMatrix(Entity* player, u
 
     std::unordered_map<LOT, int32_t> drops;
 
-    if (missionComponent == nullptr) {
-        return drops;
-    }
-
```

Comment:
Does it work without this change, given the comment by @Wincent01 on #313?

Reply:
I'll have to test without it. It seems to me that either way it is probably safer to have the checks for mission components being null to only affect the parts of loot that require it. Through testing I did confirm that this part did not affect the drops for powerup spawners, but figured I would keep it since it seemed like a safer option to me.